### PR TITLE
updated gitignore to ignore files created by local python installation

### PR DESCRIPTION
**Pull request by jameslamb**
I followed the instructions for building the Python package, and found that some of the files and directories produced are not covered by `.gitignore`. Would like to propose adding them.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -374,11 +374,16 @@ lightgbm.model
...
 # VSCode
 .vscode
 
+# Files from local Python install
+python-package/LICENSE
+python-package/build_cpp/
+python-package/compile/
+
 # duplicate version file
 python-package/lightgbm/VERSION.txt
```
Comment (StrikerRUS) [inline .gitignore]: I suppose `VERSION.txt` here is from local Python install too and can be moved under the same section.

Comment (jameslamb) [reply .gitignore]: sure that's fair. I can do that

Comment (StrikerRUS) [reply .gitignore]: Thanks!

Comment (jameslamb) [reply .gitignore]: fixed

**Comment by jameslamb** [comment]
ugh that's annoying. Should I try to find it and open another PR to fix the link? (assuming it moved and wasn't just totally removed)

**Comment by StrikerRUS** [comment]
Very often what is needed is just to wait (... and rerun the test :-) ). But I see `410` error for the first time!
Also, I'm not sure that there is a mirror for that blog post...

**Comment by jameslamb** [comment]
When I navigate to https://blogs.technet.microsoft.com/windowshpc/2015/02/02/how-to-compile-and-run-a-simple-ms-mpi-program/ I see this in my browser (I did not View Source, this is literal text being returned and not rendered):

image

**Comment by jameslamb** [comment]
```
Jamess-MacBook-Pro:LightGBM jlamb$ curl -i https://blogs.technet.microsoft.com/windowshpc/2015/02/02/how-to-compile-and-run-a-simple-ms-mpi-program/
HTTP/1.1 410 Gone
Accept-Ranges: bytes
Content-Type: text/html
ETag: "ebd0540fb44f9f92b5ca37917de3309b:1541630414.543567"
Last-Modified: Wed, 07 Nov 2018 22:40:14 GMT
Server: AkamaiNetStorage
Content-Length: 12273
Date: Thu, 21 Mar 2019 01:23:13 GMT
Connection: keep-alive
```

wild 😂

**Comment by StrikerRUS** [comment]
Hm, nothing happened, still `410`...

OK, we can use cache: https://webcache.googleusercontent.com/search?q=cache:x17s3iYl3QkJ:https://blogs.technet.microsoft.com/windowshpc/2015/02/02/how-to-compile-and-run-a-simple-ms-mpi-program/

Also ping @chivee and @guolinke for their opinions

**Comment by StrikerRUS** [comment]
Moreover, http://research.microsoft.com/en-us/projects/mslr/ is `404` now.

**Comment by guolinke** [comment]
@StrikerRUS mslr is good now.
For msmpi tutorial, maybe we can use some video contents, such as https://www.youtube.com/watch?v=iqzXhp5TxUY ?

**Comment by StrikerRUS** [comment]
@guolinke Hmm, interesting option! The only disadvantage I see is that you cannot copy-paste a code from a video. But since the code in the tutorial was represented as a screenshot, I think it's OK to provide a video.

**Comment by jameslamb** [comment]
just rebased to master now that #2065 was merged (thanks again @StrikerRUS )

**Pull request merged**